src/Microsoft.Diagnostics.Runtime/Interfaces/IClrThread.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Diagnostics.Runtime.Interfaces
+{
+    public interface IClrThread
+    {
+        ulong Address { get; }
+        IClrAppDomain? CurrentAppDomain { get; }
+        IClrException? CurrentException { get; }
+        GCMode GCMode { get; }
+        bool IsAlive { get; }
+        uint LockCount { get; }
+        int ManagedThreadId { get; }
+        uint OSThreadId { get; }
+        IClrRuntime Runtime { get; }
+        ulong StackBase { get; }
+        ulong StackLimit { get; }
+        ClrThreadState State { get; }
+
+        IEnumerable<IClrStackRoot> EnumerateStackRoots();
+        IEnumerable<ClrStackFrame> EnumerateStackTrace(bool includeContext = false);
```

Comment:
CLrStackFrame is fine so long as you allow subclasses (no interface needed).

Reply:
Ah, we do need an interface.